x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -413,17 +419,48 @@ export function registerIndexRoutes({
       const {
         model_id: modelId,
         pipeline_name: pipelineName,
+        pipeline_definition: pipelineDefinition,
         source_field: sourceField,
         destination_field: destinationField,
         inference_config: inferenceConfig,
       } = request.body;
 
+      // additional validations
+      if (pipelineDefinition && (sourceField || destinationField || modelId)) {
+        return createError({
+          errorCode: ErrorCode.PARAMETER_CONFLICT,
+          message: i18n.translate(
+            'xpack.enterpriseSearch.server.routes.createMlInferencePipeline.ParameterConflictError',
+            {
+              defaultMessage:
+                'pipeline_definition should only be provided if source_field and destination_field and model_id are not provided',
+            }
+          ),
+          response,
+          statusCode: 400,
+        });
+      } else if (!(pipelineDefinition || (sourceField && modelId))) {
+        return createError({
+          errorCode: ErrorCode.PARAMETER_CONFLICT,
+          message: i18n.translate(
+            'xpack.enterpriseSearch.server.routes.createMlInferencePipeline.ParameterMissingError',
+            {
+              defaultMessage:
+                'either pipeline_definition or source_field AND model_id must be provided',
+            }
+          ),
+          response,
+          statusCode: 400,
+        });
+      }
+
       let createPipelineResult: CreateMlInferencePipelineResponse | undefined;
```

Comment:
nitpick: Not your code, but we don't really need a let here. We can just create this variable in the `try` block and do the final return that's currently on line 488 inside the same `try` block.